### PR TITLE
fix: auto-promote next card as default when default card is deleted

### PR DESCRIPTION
**Pull request by anusuyasri**
## Summary

Fixes the DELETE /:id card endpoint which had no logic to handle deletion of the default card. When a user deleted their default card, no other card was promoted — leaving them with no default card and breaking QR code sharing. This PR adds auto-promotion logic so the next oldest card is automatically set as default.

Closes #170 

---

## Type of Change

- [x] Bug fix

---

## What Changed

- apps/backend/src/routes/cards.ts — Added a check in DELETE /:id handler: if the deleted card was the default, the next oldest card belonging to the user is automatically promoted as the new default
- Handles edge case gracefully where the deleted card is the user's last card (no promotion needed)
---

## How to Test

1.Create two cards for a user — set one as default
2.Send DELETE /cards/:id for the default card
3.Verify the remaining card is now automatically marked as isDefault: true
4.Repeat with only one card — verify deletion succeeds with no errors

---

## Checklist

- [x] My code follows the project's coding style (`pnpm -r run lint` passes).
- [x] TypeScript compiles without errors (`pnpm -r run typecheck`).
- [x] I have added or updated tests for the changes I made.
- [x] All tests pass locally (`pnpm -r run test`).
- [x] I have updated documentation where necessary.
- [x] No new `console.log` or debug statements left in the code.
- [x] Breaking changes are documented in this PR description.

---

## Screenshots / Recordings

N/A backend change logic

---

## Additional Context

The fix uses the same Prisma patterns already established in the codebase. The existing variable (already fetched before deletion for the 404 check) is reused to check isDefault, so no extra DB query is added before the delete.

### CODE REVIEW

File: apps/backend/src/routes/cards.ts [hunk 1]
```diff
@@ -144,7 +144,22 @@ export async function cardRoutes(app: FastifyInstance) {
...
+
+  if (nextCard) {
+    await app.prisma.card.update({
+      where: { id: nextCard.id },
+      data: { isDefault: true },
+    });
+  }
+}
+
+return reply.status(204).send();
```
Comment (Harxhit) [inline apps/backend/src/routes/cards.ts]: Error handling for DB logic can be better.

Comment (anusuyasri) [reply apps/backend/src/routes/cards.ts]: Thanks for the feedback, I've wrapped the DB logic in a try/catch block and return a 500 with a descriptive error message if anything fails.

File: apps/backend/src/routes/cards.ts [hunk 2]
```diff
@@ -143,8 +143,27 @@ export async function cardRoutes(app: FastifyInstance) {
...
+    try {
+      await app.prisma.card.delete({ where: { id } });
+
+      if (existing.isDefault) {
+        const nextCard = await app.prisma.card.findFirst({
+          where: { userId },
+          orderBy: { createdAt: 'asc' },
+        });
+
+        if (nextCard) {
```
Comment (Harxhit) [inline apps/backend/src/routes/cards.ts:155]: This is a db logic too?

Comment (anusuyasri) [reply apps/backend/src/routes/cards.ts:155]: yes
